README_en.md

```diff
@@ -0,0 +1,186 @@
+
+$client = new UnivapayClient(AppJWT::createToken('token', 'secret'));
+
+// For more options, create and modify the client options object before instantiating the client
+// See UnivapayClientOptions for full options list
+$clientOptions = new UnivapayClientOptions();
+$clientOptions->rateLimitHandler = new RateLimitHandler(5, 2);
+$client = new UnivapayClient(AppJWT::createToken('token', 'secret'), $clientOptions);
+
+// See the examples folder for usage examples
+```
+
+### Application Tokens
+
+Both store and merchant type application tokens are supported by this SDK. Apart from creating transaction tokens and charges which require a store type token, all other features are supported by both token types.
+
+### Money models
+This SDK uses the `moneyphp` library to model amounts and currency. Please refer to the [documentation](http://moneyphp.org/en/latest/index.html) for more details.
+All currencies and amounts will be automatically converted to `Currency` and `Money` objects. Only formatted amounts (denoted by the `.*Formatted` key) will be in string form.
+
+```php
+use MoneyCurrency;
+use MoneyMoney;
+use UnivapayPaymentMethodCardPayment;
+
+$paymentMethod = new CardPayment(...);
+$charge = $client
+    ->createToken($paymentMethod)
+    ->createCharge(Money::USD(1000));
+
+$charge->currency === new Currency('USD'); // true
+$charge->requestAmount === new Money(1000, $charge->currency); // true
+```
+
+### Enumerators
+
+As PHP has no native built in enumeration support, we provide a class called `TypedEnum` to provide type safety when working with enumerators. Each enumerator class is final and extends `TypedEnum` to provide static functions that operate similarly to enumerators in other languages like Java. A enum classes can be found in the `UnivapayEnums` namespace.
+
+_By default, if the value is not specified during creation, it will be snake-cased from the name_
+
+```php
+use UnivapayEnumsChargeStatus;
+
+$values = ChargeStatus::findValues(); // Get a list of all names and values in the enumerator
+$chargeStatus = ChargeStatus::PENDING(); // Note the braces at the end
+$chargeStatus->getValue() === 'pending'; // true
+$chargeStatus === ChargeStatus::fromValue('pending'); // true
+// Also works for switch statements
+switch ($chargeStatus) {
+    case ChargeStatus::PENDING():
+        // Do something
+        break;
+    // ...
+}
+```
+
+### Updating resource models
+To update/refresh any resource models (model classes that extends `Resource`)
+
+```php
+$charge->fetch();
+```
+
+### Long polling
+The following resources supports long polling to wait for the next status change:
+- `Charge`
+- `Refund`
+- `Cancel`
+- `Subscription`
+
+This is useful since these requests initially returns with a `PENDING` status. Long polling allows you to fetch the updated model when the resource has changed its status. If no changes occurs within 5 seconds, it will return the resource at that state.
```

Comment:
```suggestion
This is useful since these requests initially returns with a `PENDING` status. Long polling allows you to fetch the updated model when the resource has changed its status. If no changes occurs within 3 seconds, it will return the resource at that state.
```